### DISCUSSION TITLE
Grouped & Stacked bar chart?

### DISCUSSION THREAD

**Discussion by robinsandstrom** [f5/unovis#428] [Q&A]
Hi, is there any way of reproducing something like [this](https://www.chartjs.org/docs/latest/samples/bar/stacked-groups.html)?

<img width="869" alt="Screenshot 2024-08-12 at 17 25 50" src="https://github.com/user-attachments/assets/58795d86-7588-4a22-bd86-9fb26b5d6ad1">

Any help with implementation is very much appreciated

**Comment by rokotyan** [comment]
@robinsandstrom Currently Unovis doesn't support this kind of bar charts. If you would like to contribute and help us add it, we can help you set up the development environment and point you to the right place in the code.

**Reply by robinsandstrom** [reply]
Thanks! Feel free to point me in the right direction. For anyone else noticing this comment a monkey patch is just adding stacking the series like this:

```
<VisXYContainer :height="600" :data="data">
      <VisGroupedBar :data="chartData" :x="d => d.x" :y="bar1" :color="(_d, i ) => ['red', 'green'][i]" :bar-padding="0.3" />
      <VisGroupedBar :data="chartData" :x="d => d.x" :y="bar2"  :color="(_d, i ) => ['white', 'black][i]" :bar-padding="0.3" />
      <VisAxis type="y" />
    </VisXYContainer>
```

where:

```

data = [
{
  series1stack1: 5,
  series2stack1: 10,
  series1stack2: 2,
  series2stack2: 3
}...]

const bar 1 = [
 (d) => d.series1stack1,
 (d) => d.series2stack1,
]

const bar2 = [
 (d) => d.series1stack1 + d.series1stack2,
 (d) => d.series2stack1 + d.series2stack2,
]

```

which simulates a stacking like this:

<img width="376" alt="Screenshot 2024-08-13 at 21 22 43" src="https://github.com/user-attachments/assets/b54d131b-98b7-4f26-abbe-abc5a29c5a21">

:)

**Comment by rokotyan** [comment]
I would try to modify the [StackedBar](https://github.com/f5/unovis/blob/main/packages/ts/src/components/stacked-bar/index.ts) component to make it work with user-defined stacks (like the charts.js example you've shared). The code there gets a little tricky because we support Horizontal and Vertical orientations, but essentially you'll only need to modify the [`_render`](https://github.com/f5/unovis/blob/2c9925a2dc44d34c6773727f6393be2594b8c66e/packages/ts/src/components/stacked-bar/index.ts#L83) function, and maybe some of the utils.

It should be pretty easy to run the dev app locally, just clone the repo, run `npm install` and then `npm run dev`.
